### PR TITLE
Add keyword in 144

### PR DESCRIPTION
**Pull request by Incarnation-p-lee**

### CODE REVIEW

File: src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/query/DocumentQuery.java [hunk 1]
```diff
@@ -55,7 +55,12 @@ private boolean isCrossPartitionQuery(@NonNull String keyName) {
 
         final Optional<Criteria> criteria = this.getSubjectCriteria(this.criteria, keyName);
 
-        return criteria.map(criteria1 -> criteria1.getType() == CriteriaType.IS_EQUAL).orElse(true);
+        return criteria.map(criteria1 -> criteria1.getType() != CriteriaType.IS_EQUAL).orElse(true);
```
Comment (sophiaso) [inline src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/query/DocumentQuery.java]: For case isEqual and other condition, seems the result is opposite? e.g., `findByNameAndAgeGreaterThan`, will return `true`, but actually should be `false`?

File: src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/query/DocumentQuery.java [hunk 2]
```diff
@@ -55,7 +55,12 @@ private boolean isCrossPartitionQuery(@NonNull String keyName) {
...
+    }
+
+    private boolean hasKeywordOr() {
+        // If there is OR keyword in DocumentQuery, the top node of Criteria must be OR type.
+        return this.criteria.getType() == CriteriaType.OR;
```
Comment (sophiaso) [inline src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/query/DocumentQuery.java]: Why not need to check the `subCriteria` inside the `criteria` ?

File: src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/query/DocumentQuery.java (hunk 1 above)
Comment (Incarnation-p-lee) [reply src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/query/DocumentQuery.java]: typo from last PR, should be `!=` here. Yes, the result is opposite. If not equal and OR keyword, should enable CrossPartitionKey.

File: src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/query/DocumentQuery.java (hunk 2 above)
Comment (Incarnation-p-lee) [reply src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/query/DocumentQuery.java]: Actually, here we check is there any OR keyword in criteria. Any OR keyword will make the semantic to query cross ParititonKey.

**Pull request merged**